src/org/etools/j1939_84/modules/SupportedSpnModule.java

```diff
@@ -170,6 +257,78 @@ public boolean validateDataStreamSpns(ResultsListener listener,
         return result;
     }
 
+    private static Collection<SpnGroup> getDesiredDataStreamSpns(FuelType fuelType, int engineModelYear) {
+        Collection<SpnGroup> requiredSpns = new HashSet<>(DESIRED_DATA_STREAM_SPNS);
+
+        if (fuelType.isCompressionIgnition()) {
+            if (engineModelYear >= 2024) {
+                requiredSpns.addAll(DESIRED_2024_CI_DATA_STREAM_SPNS);
+            }
+        }
+        if (fuelType.isElectric()){
+            if (engineModelYear >= 2024){
+                requiredSpns.addAll(DESIRED_2024_XEV_DATA_STREAM_SPNS);
+            }
+        }
+        if (fuelType.isHybrid()){
+            if (engineModelYear >= 2024){
+                requiredSpns.addAll(DESIRED_2024_HEV_DATA_STREAM_SPNS);
+            }
+        }
+        if (engineModelYear >= 2016) {
+            requiredSpns.addAll(DESIRED_2016_DATA_STREAM_SPNS);
+        }
+        if (engineModelYear >= 2022) {
+            requiredSpns.addAll(DESIRED_2022_DATA_STREAM_SPNS);
+        }
+        if (engineModelYear >= 2024) {
+            requiredSpns.addAll(DESIRED_2024_DATA_STREAM_SPNS);
+        }
+
+        return requiredSpns.stream().sorted().collect(Collectors.toList());
+    }
+
+    public boolean validateDesiredDataStreamSpns(ResultsListener listener,
```

Comment:
I don't see where this method is being used outside of the test.  Shouldn't there be a controller class that's calling the new methods? And shouldn't there be other changes for the changing of the other method name?